packages/flutter/lib/src/material/tab_bar_theme.dart

```diff
@@ -0,0 +1,60 @@
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'tabs.dart';
+
+/// Custom theme for [TabBar].
+///
+/// Used to customize the colors of the label, and the size/shape of the
+/// indicator.
+class TabBarTheme extends Diagnosticable {
+  /// Creates a tab bar theme that overrides the defaults for the following
+  /// values for any [TabBar] in the context.
+  const TabBarTheme({
+      this.indicator,
+      this.indicatorSize,
```

Comment:
Good question...

- Default values: I'm concerned about adding any of these, because it may lead to unintended overriding of properties. For example, if the user wants to theme the label color, they may want the indicator size to default to whatever it is when `TabBar` gets a null `indicatorSize`. If I provide a default it will override this behavior.
- Null assertions: All of these properties are nullable on the `TabBar` itself, so I'm not to concerned about this. The only weird behavior I can think of is if all fields were null, but even then I don't think I want to explicitly forbid that.

I'm still new to this though so I'm open to suggestions here, let me know if you think of anything.

Reply:
Agreed!